terraform/modules/content_delivery_network/lambda_functions/redirect_s3_404/src/lambda_function.py

```diff
@@ -1,12 +1,23 @@
 # mypy: ignore-errors
+from urllib.parse import urlparse, urlunparse
 
 
 def handler(event, context):
-    """
-    This function updates the HTTP status code in the response to 307, to redirect to another
-    path (cache behavior) that has a different origin configured. Note the following:
+    """This function updates the HTTP status code in the response to 307, to
+    redirect to another path (cache behavior) that has a different origin
+    configured.
+
+    Note the following:
     1. The function is triggered in an origin response
     2. The response status from the origin server is an error status code (404)
+
+    The pattern for the incoming request uri:
+
+    /{dataset}/{version}/{implementation}/{z}/{x}/{y}.(png|pbf)
```

Comment:
Can you add the preceding segment `tiles.globalforestwatch.org` or `{tile_server_domainname}` here in this comment.  Otherwise, I think a code reader may still be confused why there are seven segments (not 6) and implementation is at index 3 rather than index 2.

Reply:
Oh, i see the confusion about the confusion, @danscales. 👍 
I added clarification in 4f08c6802f51d3b9c87fd474b2d5822a711afff8

The domain isn't part of the URL from the perspective of the handler. Instead, the URI and redirect URL are [relative](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location#directives). 

I added this screenshot of the network traffic to the PR:
![redirect_issue](https://github.com/user-attachments/assets/43044ca3-68d9-4773-9216-ad3b956456e3)